commit f117c7d94a2abb0561540a1fe33dd729b1cb8b34
Author: nicm <nicm>
Date:   Sun Jul 13 20:23:10 2014 +0000

    If a client is killed while suspended with ^Z so has gone through the
    MSG_EXITED dance, don't try to resume it since a) it's pointless and b)
    the tty structures have been cleaned up and tmux will crash.

diff --git a/server-client.c b/server-client.c
index 0f4d39d..1ef3f31 100644
--- a/server-client.c
+++ b/server-client.c
@@ -875,6 +875,9 @@ server_client_msg_dispatch(struct client *c)
 				break;
 			c->flags &= ~CLIENT_SUSPENDED;
 
+			if (c->tty.fd == -1) /* exited in the meantime */
+				break;
+
 			if (gettimeofday(&c->activity_time, NULL) != 0)
 				fatal("gettimeofday");
 			if (c->session != NULL)
